Client/SafeBrowsing/GSafeBrowsingDatabase.swift

```diff
@@ -115,6 +137,44 @@ class SafeBrowsingDatabase {
         }
     }
     
+    func canFind() -> Bool {
+        return Date().timeIntervalSince(self.lastFindDate) >= self.lastFindWaitDuration
+    }
+    
+    func canUpdate() -> Bool {
+        return Date().timeIntervalSince(self.lastFetchDate) >= self.lastFetchWaitDuration
+    }
+    
+    func enterBackoffMode(_ mode: BackoffMode) {
```

Comment:
I think we can simplify this by keeping a single backoff flag for both endpoints. It's not 100% clear in
the [spec](https://developers.google.com/safe-browsing/v4/request-frequency#back-off-mode) but keeping a single flag would be the safest interpretation and the easiest implementation.

Reply:
Yes but when I read the spec, I came across: 

> For example, if a fullHashes response contains a minimum wait duration of 1 hour, the client must not send send any fullHashes requests until that hour passes.

For minimum duration and since it explicitly states `must not send "fullHashes"`, I assume it's only for fullHashes. I made the assumption that it's the same for both calls as they relate to each other.

If update fails, we can still consult the cache and `find` endpoint.

Reply:
Back-off is only triggered when the server is overloaded or when a client is misbehaving. In other words, the server is telling the client to go away and so I think it would be preferable to stay away from the server entirely.

You're right that it's not ideal for security since we'll be unable to confirm results and might miss bad URLs and that's another example where keeping the server alive is more important than ensuring that each client receives the response it needs. Thankfully it's a rare occurence in my experience (with Firefox).